### PR TITLE
Small cleanup of ArchetypeType handling in substType()

### PR DESCRIPTION
**Pull request by slavapestov**
Should be NFC.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

File: lib/AST/Type.cpp [hunk 1]
```diff
@@ -4390,6 +4390,19 @@ static Type getMemberForBaseType(LookupConformanceFn lookupConformances,
...
+    return getDependentMemberType(substBase);
+
+  // All remaining cases require an associated type declaration and not just
+  // the name of a member type.
+  if (!assocType)
+    return failed();
+
   // If the parent is an archetype, extract the child archetype with the
   // given name.
   if (auto archetypeParent = substBase->getAs<ArchetypeType>()) {
```
Comment (AnthonyLatsis) [inline lib/AST/Type.cpp:4413]: Does the `archetypeParent->getNestedTypeByName` case require or imply that `assocType` is non-null too?

File: lib/AST/Type.cpp [hunk 2]
```diff
@@ -4411,7 +4411,7 @@ static Type getMemberForBaseType(LookupConformanceFn lookupConformances,
...
-    if (Type memberArchetypeByName = archetypeParent->getNestedTypeByName(name))
+    if (Type memberArchetypeByName = archetypeParent->getNestedType(assocType))
```
Comment (AnthonyLatsis) [inline lib/AST/Type.cpp:4414]: Ah, nvm.

**Pull request merged**